src/hotspot/share/gc/shenandoah/shenandoahFreeSet.cpp

```diff
@@ -439,6 +422,12 @@ void ShenandoahFreeSet::flip_to_gc(ShenandoahHeapRegion* r) {
     adjust_bounds();
   }
   assert_bounds();
+
+  shenandoah_assert_heaplocked();
+  try_recycle_trashed(r);
+
+  assert(r->is_empty(), "Region must be empty after flipping from mutator to GC.");
+  r->set_affiliation(FREE);
```

Comment:
Why is this block needed now?

Reply:
I think this could be an assert. The region should already be `FREE` at this point.

Reply:
When debugging, I came across regions that were "trash" but not "FREE".

Reply:
Turns out that was OK, because the code at hand here is actually always followed by code that does the same (in try_allocate_in()). My code insertion here is thus redundant. I will remove it.